pkg/ruler/ruler.go

```diff
@@ -266,32 +255,12 @@ func (r *Ruler) starting(ctx context.Context) error {
 // Stop stops the Ruler.
 // Each function of the ruler is terminated before leaving the ring
 func (r *Ruler) stopping(_ error) error {
-	r.notifiersMtx.Lock()
-	for _, n := range r.notifiers {
-		n.stop()
```

Comment:
This was done before stopping the subservices. I believe you should run `r.manger.Stop()` before stopping the subservices. WDYT?

Reply:
I agree with you. I will move it.